File: backend/src/CloudCrafter.Domain/Domain/Stacks/StackEnvironmentVariableHistoryDto.cs
using CloudCrafter.Domain.Entities;

namespace CloudCrafter.Domain.Domain.Stacks;

public class StackEnvironmentVariableHistoryDto
{
    public Guid Id { get; set; }
    public Guid StackId { get; set; }
    public Guid EnvironmentVariableId { get; set; }
    public string Key { get; set; } = string.Empty;
    public string? OldValue { get; set; }
    public string? NewValue { get; set; }
    public bool IsSecret { get; set; }
    public EnvironmentVariableType Type { get; set; }
    public string ChangeType { get; set; } = string.Empty; // Created, Updated, Deleted
    public string? GroupName { get; set; }
    public DateTime Timestamp { get; set; }
    public string? UserId { get; set; }
    public string? UserName { get; set; }
}

File: frontend/packages/api/src/__generated__/types/StackEnvironmentVariableHistoryDto.ts
import type { EnvironmentVariableType } from './EnvironmentVariableType'

export type StackEnvironmentVariableHistoryDto = {
  /**
   * @type string, uuid
   */
  id: string
  /**
   * @type string, uuid
   */
  stackId: string
  /**
   * @type string, uuid
   */
  environmentVariableId: string
  /**
   * @type string
   */
  key: string
  /**
   * @type string
   */
  oldValue?: string | null
  /**
   * @type string
   */
  newValue?: string | null
  /**
   * @type boolean
   */
  isSecret: boolean
  /**
   * @type integer
   */
  type: EnvironmentVariableType
  /**
   * @type string
   */
  changeType: string
  /**
   * @type string
   */
  groupName?: string | null
  /**
   * @type string, date-time
   */
  timestamp: string
  /**
   * @type string
   */
  userId?: string | null
  /**
   * @type string
   */
  userName?: string | null
}